src/main/scala/com/datastax/sparkstress/ReadTask.scala

```diff
@@ -48,18 +59,169 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
+      .load()
+      .select("color")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Two Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSTwoColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Three Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSThreeColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size", "qty")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Four Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSFourColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size", "qty", "order_number")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Five Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSFiveColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("order_number", "qty", "color", "size", "order_time")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan All Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSAllColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format ("org.apache.spark.sql.cassandra")
+      .options (Map ("table" -> table, keyspace -> "keyspace") )
+      .load()
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Read Parquet file from DSEFS using Datasets.
+  */
+class DSEFSReadParquet_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read.parquet(s"dsefs:///${config.keyspace}.${config.table}").count
```

Comment:
My understanding is that XYZTask is executed more than once. Do we somehow consider various caches interfering with this test?

Reply:
We've been dealing with caches outside spark stress. We first load the data, flush all memtables, clear all system caches, then run the read test and monitor various metrics. Caches will be a bigger deal if we run multiple trials, but I usually stay away from using that feature for that exact reason.